Nueva base de datos

leonel-cc opened the pull request:
se refactorizaron funciones ya existentes para poder implementar la nueva base de datos

Endpoint -> POST "http://localhost:3000/admin-products"

Json
{
"url":"https://docs.google.com/spreadsheets/d/1Og282LpUBbJW0bsMYGmiePeCLu7P0MGN/edit#gid=1648496528"
}

leonel-cc committed:
Merge branch '168-crear-unit-tests-para-el-módulo-de-las-reviews' of https://github.com/Actualiza-Tu-Carro/back-end-atc into grimreaper

leonel-cc committed:
Merge branch '164-crear-método-para-realizar-pago' of https://github.com/Actualiza-Tu-Carro/back-end-atc into grimreaper

leonel-cc committed:
Merge branch 'development' of https://github.com/Actualiza-Tu-Carro/back-end-atc into grimreaper

leonel-cc committed:
Nueva base de datos

se refactorizaron funciones ya existentes para poder implementar
la nueva base de datos

lechugaxthz commented on src/admin-products/admin-products.service.ts:
```diff
@@ -92,8 +92,38 @@ export class AdminProductsService {
-          resolve(result.data);
+          // Mapeo de nombres de propiedades
+          const propertyMapping = {
+            '': 'Número de publicación',
```
por eso decia que le ponga encabezado al excel

lechugaxthz commented on src/products/entities/product.entity.ts:
```diff
@@ -16,8 +16,8 @@ import { ShoppingCart } from '../../shopping-cart/entities/shopping-cart.entity'
 export enum stateproduct {
-  Active = 'Activa',
-  Inactive = 'Inactiva',
+  Nuevo = 'Nuevo',
```
consulto por el sentido de éste, porque antes "significaba otra cosa"

lechugaxthz commented on src/products/products.service.ts:
```diff
@@ -301,17 +301,13 @@ export class ProductsService {
         id: product['Número de publicación'],
         title: product.Título,
-        description: product.Descripción,
+        description: product['Descripción'],
```
descripción  te lo debería tomar de todos modos. si es una string concatenada, ahi si el `[' ']`, pero funciona igual

lechugaxthz commented on src/products/products.service.ts:
```diff
@@ -321,7 +317,7 @@ export class ProductsService {
-        } del indice ${index + 2}`,
+        } del indice ${index + 2}` + error.message,
       );
     }
```
ten en cuenta que el mensaje sería lo que está dentro de la ' ' (en realidad son los backsticks pero si pongo, sale el otro tipo de texto. lo digo porque estas mandando 2 propiedades message por lo que entiendo, a lo sumo manda antes del ultimo backstick un \n ${error.message}

leonel-cc committed:
base de datos nueva 1.1

El atributo state volvio a ser el anterior que especifica si un producto esta activo o no.
Se implemento el atributo condicion que determina si un producto es nuevo, remanufacturado o usado.
Ademas se cambio la forma en la que se guardan las imagenes, en un array de string.

leonel-cc committed:
base de datos nueva 1.2

Mejore la forma en la que se separan las imagenes utilizando .split(",")

leonel-cc committed:
Merge branch '192-databasenew' of https://github.com/Actualiza-Tu-Carro/back-end-atc into grimreaper

leonel-cc committed:
Merge branch development and databasenew

leonel-cc committed:
base de datos nueva 1.3

se implemento un nuevo atributo a los productos, llamado "mostSelled"
de tipo booleano.
este atributo sirve para diferenciar los productos mas vendidos de los
que menos se venden y asi poder filtrarlos.

se creo una nueva ruta de productos
PATCH: "http://localhost:3000/products/mostSelled/:idDelProducto"

mediante esta ruta podemos cambiar el valor booleano de mostSelled a True
si esta en false y viceversa.

agregue un Guard a esta ruta para que solo pueda acceder el administrador,
pero por el momento lo dejo comentado para que puedan hacer pruebas.

leonel-cc commented:
se implemento un nuevo atributo a los productos, llamado "mostSelled"
de tipo booleano.
este atributo sirve para diferenciar los productos mas vendidos de los
que menos se venden y asi poder filtrarlos.

se creo una nueva ruta de productos
PATCH: "http://localhost:3000/products/mostSelled/:idDelProducto"

mediante esta ruta podemos cambiar el valor booleano de mostSelled a True
si esta en false y viceversa.

agregue un Guard a esta ruta para que solo pueda acceder el administrador,
pero por el momento lo dejo comentado para que puedan hacer pruebas.
